interfaces/builtin/utils.go

```diff
@@ -102,3 +104,19 @@ func implicitSystemConnectedSlot(slot *interfaces.ConnectedSlot) bool {
 	}
 	return false
 }
+
+// determine if the given slot attribute path matches the regex
+func verifySlotPathAttribute(slotRef *interfaces.SlotRef, attrs interfaces.Attrer, reg *regexp.Regexp, errStr string) (string, error) {
+	var path string
+	if err := attrs.Attr("path", &path); err != nil || path == "" {
+		return "", fmt.Errorf("slot %q must have a path attribute", slotRef)
+	}
+	cleanPath := filepath.Clean(path)
+	if cleanPath != path {
+		return "", fmt.Errorf(`cannot use slot %q path %q: try %q"`, slotRef, path, cleanPath)
+	}
+	if !reg.MatchString(cleanPath) {
+		return "", fmt.Errorf("slot %q path attribute %s", slotRef, errStr)
```

Comment:
Minor nitpick, most likely this cannot ever be translated correctly because grammar elements in the fixed element here will have to agree on case (and many other properties) with the `errString` text which translators won't be able to get right if the string in `errString` has a separate standalone use case that needs a different set of properties.

In general in i18n it's incorrect to do `_("hello") + " " + _("world")`. While tedious you must do `_("hello world")` or introduce a layer where linguistic properties can be expressed so that both `"hello"` and `"world"` can have multiple translations depending on the context. It's usually avoided by using tricks like "error: ..." where the users are more accustomed to disagreeing linguistic properties of both parts. It's much more striking and plain wrong if it is done in one sentence.